### ISSUE TITLE
Glitched Case/Map

### ISSUE THREAD

**Issue by KillaBoi** [BeepIsla/CSGO-Overwatch-Bot#178]
Heyya,

So a friend of mine got this issue whilst doing an Overwatch case, seems like he got a case that was old (back from CS:GO Version 1.36.7.6 which is approximately 2 years old, [Closest SteamDB Changelog](https://steamdb.info/app/730/history/?changeid=5719004)) for some reason and neither the actual Overwatch Case in game worked and the bot crashed as well.

![image](https://i.imgur.com/ny3OO9U.png)

By manually downloading the Demo from the URL and playing it within the game, we can see the following error in console:

```
WARNING: demo network protocol 13676 differs, engine version is 13770 
Playing demo from C:\Users\<REDACTED>\Downloads\003327729322227138561_1182326985.dem.
WARNING: Server demo version 13676, client version 13770.
Map version fallback for de_dust2 CRC=1357023846: 1773333823 (CRC=2613617750)
```
and it proceeds to download the map from the workshop. Perhaps this is linked in with the China CS:GO issues we are facing as well since some of the maps are not the same as the official maps we get in our versions due to censorship?

```
Download URL: http://replay202.valve.net/730/003327729322227138561_1182326985.dem.bz2 // https://drive.google.com/file/d/1ovXIf2cL0LFP0bXT6SLakxNdG4v2uews/view
Total Rounds: 21
Rounds to watch: 8
Overwatch starting round: 14
Suspect: https://steamcommunity.com/profiles/76561198843209523
```

**Comment by KillaBoi** [comment]
PS, here's the link to the actual map the game downloads from the workshop: https://steamcommunity.com/sharedfiles/filedetails/?id=1773333823

**Comment by GitHubberDEN** [comment]
Same here 
![image](https://user-images.githubusercontent.com/22105813/98372869-b9ab3b00-203e-11eb-9c8b-584e1e5da2dd.png)

**Comment by Faltez** [comment]
having the same issue while running the bot when downloading the case and unpacking bot crashes and i receive same errors

**Comment by BeepIsla** [comment]
1. I doubt you all are receiving that exact error with old demos, this is the first time I ever heard about this. You most likely are just getting cases from Perfect World which have been an issue for a while (See: #30)

2. Nothing I can really do about it, just wait 90 minutes and try again, you typically get a new case after that time.

**Comment by Faltez** [comment]
![image](https://user-images.githubusercontent.com/73907827/98420406-3a1f7980-20db-11eb-8688-8ef67a1342f7.png)

**Comment by Faltez** [comment]
thats the error im getting

**Comment by Borlax2** [comment]
Same error, waiting 12 hours, tried 10 accounts and all failed at 0% parsing same as above.

0 case were solved today.

**Comment by KillaBoi** [comment]
> 1. I doubt you all are receiving that exact error with old demos, this is the first time I ever heard about this. You most likely are just getting cases from Perfect World which have been an issue for a while (See: #30)
> 2. Nothing I can really do about it, just wait 90 minutes and try again, you typically get a new case after that time.

The case I linked is from a South American MM game and it does not download from a Wanmei based server either.
![image](https://i.imgur.com/MolWyoP.png)

**Comment by Borlax2** [comment]
9|OW5      |     at new Bzip2Error (C:\Users\***\Desktop\OW\**OverWatch - 5\node_modules\unbzip2-stream\lib\bzip2.js:31:19)
9|OW5      |     at Object.Error (C:\Users\***\Desktop\OW\**OverWatch - 5\node_modules\unbzip2-stream\lib\bzip2.js:36:37)
9|OW5      |     at Object.bzip2.header (C:\Users\***\Desktop\OW\**OverWatch - 5\node_modules\unbzip2-stream\lib\bzip2.js:152:39)
9|OW5      |     at decompressBlock (C:\Users\***\Desktop\OW\**OverWatch - 5\node_modules\unbzip2-stream\index.js:18:29)
9|OW5      |     at decompressAndQueue (C:\Users\**\Desktop\OW\**OverWatch - 5\node_modules\unbzip2-stream\index.js:48:20)
9|OW5      |     at Stream.end (C:\Users\***\Desktop\OW\****OverWatch - 5\node_modules\unbzip2-stream\index.js:83:17)
9|OW5      |     at _end (C:\Users\***\Desktop\OW\****OverWatch - 5\node_modules\through\index.js:65:9)
9|OW5      |     at Stream.stream.end (C:\Users\***\Desktop\OW\**OverWatch - 5\node_modules\through\index.js:74:5)
9|OW5      |     at Readable.onend (_stream_readable.js:660:10)
9|OW5      |     at Object.onceWrapper (events.js:416:28)
9|OW5      |     at Readable.emit (events.js:310:20)
9|OW5      |     at endReadableNT (_stream_readable.js:1187:12)
9|OW5      |     at processTicksAndRejections (internal/process/task_queues.js:84:21) {
9|OW5      |   message: 'No magic number found'

**Comment by BeepIsla** [comment]
@Borlax2 Nobody here had unpacking issues, you are likely receiving cases from Perfect World. What is the download URL?

**Comment by Borlax2** [comment]
Download URL: http://replay234.wmsj.cn/730/003440271565759971351_1531951469.dem.bz2

**Comment by BeepIsla** [comment]
Unrelated to this issue. Overwatch cases from Perfect World cause these issues are reported here #30

**Comment by EV96** [comment]
> Heyya,
> 
> So a friend of mine got this issue whilst doing an Overwatch case, seems like he got a case that was old (back from CS:GO Version 1.36.7.6 which is approximately 2 years old, [Closest SteamDB Changelog](https://steamdb.info/app/730/history/?changeid=5719004)) for some reason and neither the actual Overwatch Case in game worked and the bot crashed as well.
> 
> ![image](https://camo.githubusercontent.com/dfd901d6976114cd78d94dd56281751f82e7099baf83fc4ccc24bdd764ab9f58/68747470733a2f2f692e696d6775722e636f6d2f6e79334f4f39552e706e67)
> 
> By manually downloading the Demo from the URL and playing it within the game, we can see the following error in console:
> 
> ```
> WARNING: demo network protocol 13676 differs, engine version is 13770 
> Playing demo from C:\Users\<REDACTED>\Downloads\003327729322227138561_1182326985.dem.
> WARNING: Server demo version 13676, client version 13770.
> Map version fallback for de_dust2 CRC=1357023846: 1773333823 (CRC=2613617750)
> ```
> 
> and it proceeds to download the map from the workshop. Perhaps this is linked in with the China CS:GO issues we are facing as well since some of the maps are not the same as the official maps we get in our versions due to censorship?
> 
> ```
> Download URL: http://replay202.valve.net/730/003327729322227138561_1182326985.dem.bz2 // https://drive.google.com/file/d/1ovXIf2cL0LFP0bXT6SLakxNdG4v2uews/view
> Total Rounds: 21
> Rounds to watch: 8
> Overwatch starting round: 14
> Suspect: https://steamcommunity.com/profiles/76561198843209523
> ```

Same error for me, the bot doesn't work.

**Comment by Faltez** [comment]
after posting my error i waited a day and tried again seems to be working fine now..

**Comment by KillaBoi** [comment]
> after posting my error i waited a day and tried again seems to be working fine now..

that would be because that case got dropped off your account... if we're not able to do that particular case due to a crash or something, is it possible to handle the error/exception and pause the bot until the case drops off or something instead?

**Comment by BeepIsla** [comment]
Should be fixed in 2.6.3

Just waiting 90 minutes for the case to be dropped and hopefully just continue afterwards.

Reopen if issue still occurs.
